Update sql-agent-i5O-case-study notebook (#407)
cohere-ai/cohere-developer-experience: fern/pages/cookbooks/sql-agent-cohere-langchain.mdx

    ```[0m[32;1m[1;3m
    Invoking: `sql_db_query` with `{'query': "SELECT station_name, SUM(CAST(duration AS INTEGER)) AS total_duration FROM product_tracking WHERE zone = 'wait' GROUP BY station_name ORDER BY total_duration DESC LIMIT 1;"}`
    responded: The query is correct. I will now execute it.
    
    [0m[36;1m[1;3m[('stn4', 251)][0m[32;1m[1;3mThe station with the highest total duration in the wait zone is stn4, with a total duration of 251 minutes.[0m
    
    [1m> Finished chain.[0m
    The station with the highest total duration in the wait zone is stn4, with a total duration of 251 minutes.

## Memory in the sql agent
Often times, we want the agent to hold memory of our previous messages so that we're able to coherently engage with the agent to answer our queries. In this section, let us take a look at how we can add memory to the agent so that we're able to achieve this outcome!

```python
from langchain_core.runnables.history import RunnableWithMessageHistory
from langchain_core.chat_history import BaseChatMessageHistory
from langchain_core.messages import BaseMessage
from pydantic import BaseModel, Field
from typing import List
```

> In the code snippets below, we create a class to store the chat history in memory. This can be customised to store the messages from a database, or any cache of sorts as suitable for the use case.

Reviewer: ```suggestion
In the code snippets below, we create a class to store the chat history in memory. This can be customised to store the messages from a database or any other suitable data store.
```
